fix: update ro templates

ana-yldio opened the pull request:
https://trello.com/c/5My9r4Bx/1727-2-change-detail-pages-for-presentation

jhonny-m reviewed (commented):
I think @kmaid wanted to keep only one Research Output instead of having to maintain two types, most of the work seems to be separating them

kmaid commented on apps/frontend/src/shared-research/ResearchOutput.tsx:
```diff
@@ -1,10 +1,25 @@
 import { useResearchOutputById } from './state';
 import Frame from '../structure/Frame';
 
+const getResearchOutputTemplate = (type: ResearchOutputResponse['type']) => {
```
Can we do something like it was before? https://github.com/yldio/asap-hub/pull/1150/files#diff-d532068a9bbf7564c5df0eae332d711fc99586a2084562d38e21e5e03ba71607L22. Its allot more readable

jhonny-m replied:
IMO, we could keep it as only one research output component, since most of the things are going to be the same. with the container, header contact info and just have a component for the different part with is the description tags and access information which are the differences

kmaid commented on apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx:
```diff
@@ -69,64 +69,206 @@ const renderComponent = async () => {
+    const { getByText } = await renderComponent();
+    expect(getByText(/dataset title/i)).toBeVisible();
+    expect(getByText(/dataset contents/i)).toBeVisible();
+    expect(getByText(/dataset contents/i)).toBeVisible();
```
I don't understand this assertion. its identical to the one above

kmaid commented on apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx:
```diff
@@ -69,64 +69,206 @@ const renderComponent = async () => {
   });
-  it('links to a teams', async () => {
+
+  it('renders the proposal type', async () => {
```
This is probably a regression test.

kmaid commented on apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx:
```diff
@@ -69,64 +69,206 @@ const renderComponent = async () => {
+    const { getByText } = await renderComponent();
+    expect(getByText(/protocol title/i)).toBeVisible();
+    expect(getByText(/protocol contents/i)).toBeVisible();
+    expect(getByText(/protocol contents/i)).toBeVisible();
```
Same again

kmaid commented on apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx:
```diff
@@ -69,64 +69,206 @@ const renderComponent = async () => {
+    const { getByText } = await renderComponent();
+    expect(getByText(/lab resource title/i)).toBeVisible();
+    expect(getByText(/lab resource contents/i)).toBeVisible();
+    expect(getByText(/lab resource contents/i)).toBeVisible();
```
and again

kmaid commented on apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx:
```diff
@@ -69,64 +69,206 @@ const renderComponent = async () => {
+    });
+    const { getByText } = await renderComponent();
+    expect(getByText(/bioinformatics title/i)).toBeVisible();
+    expect(getByText(/bioinformatics contents/i)).toBeVisible();
```
Can we not do this as a it.each? This is 5 duplicated tests which essentially test one prop difference.

kmaid commented on packages/react-components/src/organisms/__tests__/SharedResearchOutputHeaderCard.test.tsx:
```diff
@@ -65,14 +65,40 @@ it('shows authors', () => {
 
-it('shows labs', () => {
-  const { getByText } = render(
+it('handle labs data', () => {
```
```suggestion
it('handles labs data', () => {
```

kmaid commented on packages/react-components/src/templates/SharedResearchOutputGrantDocument.tsx:
```diff
@@ -0,0 +1,59 @@
+    backHref: string;
+  } & ComponentProps<typeof SharedResearchAdditionalInformationCard>;
+
+const SharedResearchOutputGrantDocument: React.FC<SharedResearchOutputGrantDocumentProps> =
```
I wonder if this name should reflect the difference between the templates which is an expanded description SharedResearchOutputRTFDescription

kmaid reviewed (commented):
I am not convinced that separating the templates has made this easier to read or maintain what do you think @jhonny-m ?

jhonny-m reviewed (commented):
the difference between the components seems so small that it shouldn't justify 2 different components, I would suggest pehaps separating the difference which seems to be the tags and description to a different component if we want so simplify this template

ana-yldio commented:
@kmaid @jhonny-m I'm afraid I have to disagree, tbh I think it's clearer breaking the logic into two different templates.
But since you both agree on this, I reverted it and just updated the `isGrantDocument` variable,

niezgoda commented on packages/react-components/src/templates/SharedResearchOutput.tsx:
```diff
@@ -39,7 +39,7 @@ const SharedResearchOutput: React.FC<SharedResearchOutputProps> = ({
 }) => {
   const isGrantDocument = (
-    ['Grant Document', 'Proposal'] as ResearchOutputType[]
+    ['Grant Document', 'Proposal', 'Presentation'] as ResearchOutputType[]
```
this needs to be changed after merge https://github.com/yldio/asap-hub/pull/1194
